File: kakomon10/q9.jl
function main()
	S=chomp(reverse(readline()))
	T=["dream","dreamer","erase","eraser"]

	for i =1:length(T)
		T[i]=reverse(T[i])
	end

	can=true
	x=1
	while(x<length(S))
		can2=false
		for t in T
			if occursin.(S,t)
				S=replace(S,t=>"")
				can2=true
				x+=length(t)
				println(S)
			end
		end
		if !can2
			can=false
			break
		end
	end
	println(can ? "YES" : "NO")
end

main()

File: kakomon10/py/q9.py
S = input()[::-1]
wds= "dream dreamer erase eraser"[::-1].split()

p = 0
while p < len(S):
  match = False
  for wd in wds:
    if len(S) < len(wd):
      break

    if S[p:p+len(wd)] == wd:
      p += len(wd)
      match = True
  
  if not match:
    break

if p == len(S):
  print("YES")
else:
  print("NO")
